app/ghcup/GHCup/OptParse/Config.hs

```diff
@@ -203,22 +203,40 @@ config configCommand settings userConf keybindings runLogger = case configComman
         pure $ ExitFailure 65
       VLeft _ -> pure $ ExitFailure 65
 
-  AddReleaseChannel uri -> do
-    case urlSource settings of
-      AddSource xs -> do
-        doConfig (defaultUserSettings { uUrlSource = Just $ AddSource (xs <> [Right uri]) })
-        pure ExitSuccess
-      GHCupURL -> do
-        doConfig (defaultUserSettings { uUrlSource = Just $ AddSource [Right uri] })
-        pure ExitSuccess
-      OwnSource xs -> do
-        doConfig (defaultUserSettings { uUrlSource = Just $ OwnSource (xs <> [Right uri]) })
-        pure ExitSuccess
-      OwnSpec spec -> do
-        doConfig (defaultUserSettings { uUrlSource = Just $ OwnSource ([Left spec, Right uri]) })
+  AddReleaseChannel force uri -> do
+    r <- runE @'[DuplicateReleaseChannel] $ do
+      case urlSource settings of
+        AddSource xs -> do
+          when (not force && Right uri `elem` xs) $ throwE (DuplicateReleaseChannel uri)
```

Comment:
here we error

Reply:
So, if I understand correctly, addition sequence `a.b.a` will now error unless `force`d.  
But also `a.a` will error, but this is in effect the same as `a`, because the action of adding a fixed channel at the end is idempotent (if we ignore the extra entry in `xs`).
I am a bit worried about the error in such don't-care situations.  This could lead to regressions for uses of `ghcup` in scripts.  
But I understand that just ignoring duplicate additions with a warning (unless `force`) is also not a backwards-compatible solution, and even worse than a hard error.

Could one stick with a warning just in case the last channel is added again?  In these cases we know that adding it again is redundant.